sonar-db/src/test/java/org/sonar/db/component/ComponentTesting.java

```diff
@@ -29,13 +30,13 @@
 
 public class ComponentTesting {
 
-  public static ComponentDto newFileDto(ComponentDto subProjectOrProject) {
-    return newFileDto(subProjectOrProject, Uuids.create());
+  public static ComponentDto newFileDto(ComponentDto subProjectOrProject, @Nullable ComponentDto directory) {
```

Comment:
in the vast majority of cases, we don't care about the directory. I suggest to have several constructor method, with and without the directory parameter